sre_yield/tests/test_sre_yield.py

```diff
@@ -58,6 +58,36 @@ def testAlternationWithEmptyElement(self):
         self.assertSequenceEqual(sre_yield.AllStrings("a[bc]?"), ["a", "ab", "ac"])
         self.assertSequenceEqual(sre_yield.AllStrings("a[bc]??"), ["a", "ab", "ac"])
 
+    def testHttp(self):
+        self.assertSequenceEqual(sre_yield.AllStrings("^http://"), ["http://"])
+
+    def testLookaroundNegativeNop(self):
+        self.assertSequenceEqual(sre_yield.AllStrings("(?!b)c"), ["c"])
+        self.assertSequenceEqual(sre_yield.AllStrings("(?!c)c"), ["c"])
+        self.assertSequenceEqual(sre_yield.AllStrings("(?!b)c?"), ["", "c"])
+        self.assertSequenceEqual(sre_yield.AllStrings("a(?!b)c"), ["ac"])
+        self.assertSequenceEqual(sre_yield.AllStrings("a(?!b)c?"), ["a", "ac"])
+        self.assertSequenceEqual(sre_yield.AllStrings("a(?!b)$"), ["a"])
+
+        self.assertSequenceEqual(sre_yield.AllStrings("(?<!b)c"), ["c"])
+        self.assertSequenceEqual(sre_yield.AllStrings("^(?<!b)c"), ["c"])
+        self.assertSequenceEqual(sre_yield.AllStrings("a(?<!b)c"), ["ac"])
+
+    def testLookaroundPositiveAdd(self):
+        self.assertSequenceEqual(sre_yield.AllStrings("(?<=abc)def"), ["abcdef"])
```

Comment:
This one doesn't make a sense to me if we're treating the matches as fullmatch only; the `abc` part would be before the beginning of the match.  Is there an equivalent one like `xxx(?<=abc)def` that demonstrates what you want to show as well?